server/src/main/java/com/cloud/user/AccountManagerImpl.java

```diff
@@ -2696,6 +2696,13 @@ public void buildACLSearchParameters(Account caller, Long id, String accountName
             if (id == null) {
                 domainIdRecursiveListProject.third(Project.ListProjectResourcesCriteria.SkipProjectResources);
             }
+
+            if ((caller.getType() == Account.ACCOUNT_TYPE_ADMIN) ||
+                    (caller.getType() == Account.ACCOUNT_TYPE_DOMAIN_ADMIN) ||
+                    (caller.getType() == Account.ACCOUNT_TYPE_RESOURCE_DOMAIN_ADMIN) ||
+                    (caller.getType() == Account.ACCOUNT_TYPE_READ_ONLY_ADMIN)) {
```

Comment:
We could extract all this enums to a constant list and verify if it contains the caller.getType.